x-pack/plugins/security_solution/common/detection_engine/schemas/request/get_rule_execution_events_schema.ts

```diff
@@ -7,24 +7,96 @@
+import { DefaultEmptyString, IsoDateString } from '@kbn/securitysolution-io-ts-types';
 
+import { Either } from 'fp-ts/lib/Either';
+import {
+  ExecutionLogTableSortColumns,
+  executionLogTableSortColumns,
+  ruleExecutionStatus,
+  RuleExecutionStatusType,
+} from '../common';
+
+/**
+ * Types the DefaultStatusFiltersStringArray as:
+ *   - If undefined, then a default array will be set
+ *   - If an array is sent in, then the array will be validated to ensure all elements are a ruleExecutionStatus
+ */
+export const DefaultStatusFiltersStringArray = new t.Type<
+  RuleExecutionStatusType[],
+  RuleExecutionStatusType[],
+  unknown
+>(
+  'DefaultStatusFiltersStringArray',
+  t.array(ruleExecutionStatus).is,
+  (input, context): Either<t.Errors, RuleExecutionStatusType[]> => {
+    if (input == null) {
+      return t.success([]);
+    } else if (typeof input === 'string') {
+      return t.array(ruleExecutionStatus).validate(input.split(','), context);
+    } else {
+      return t.array(ruleExecutionStatus).validate(input, context);
+    }
+  },
+  t.identity
+);
+
+/**
+ * Types the DefaultSortField as:
+ *   - If undefined, then a default sort field of 'timestamp' will be set
+ *  - If a string is sent in, then the string will be validated to ensure it is as valid sortFields
+ */
+export const DefaultSortField = new t.Type<
+  ExecutionLogTableSortColumns,
+  ExecutionLogTableSortColumns,
+  unknown
+>(
+  'DefaultSortField',
+  executionLogTableSortColumns.is,
+  (input, context): Either<t.Errors, ExecutionLogTableSortColumns> =>
+    input == null ? t.success('timestamp') : executionLogTableSortColumns.validate(input, context),
+  t.identity
+);
+
+const sortOrder = t.keyof({ asc: null, desc: null });
+type SortOrder = t.TypeOf<typeof sortOrder>;
+
+/**
+ * Types the DefaultSortOrder as:
+ *   - If undefined, then a default sort order of 'desc' will be set
+ *   - If a string is sent in, then the string will be validated to ensure it is as valid sortOrder
+ */
+export const DefaultSortOrder = new t.Type<SortOrder, SortOrder, unknown>(
+  'DefaultSortOrder',
+  sortOrder.is,
+  (input, context): Either<t.Errors, SortOrder> =>
+    input == null ? t.success('desc') : sortOrder.validate(input, context),
+  t.identity
+);
+
+/**
+ * Route Request Params
+ */
 export const GetRuleExecutionEventsRequestParams = t.exact(
   t.type({
     ruleId: t.string,
   })
 );
 
+/**
+ * Route Query Params (as constructed from the above codecs)
+ */
 export const GetRuleExecutionEventsQueryParams = t.exact(
   t.type({
-    start: t.string,
-    end: t.string,
-    query_text: t.union([t.string, t.undefined]),
-    status_filters: t.union([t.string, t.undefined]),
-    per_page: t.union([t.string, t.undefined]),
-    page: t.union([t.string, t.undefined]),
-    sort_field: sortFieldOrUndefined,
-    sort_order: sortOrderOrUndefined,
+    start: IsoDateString,
+    end: IsoDateString,
```

Comment:
Would it be helpful to have default values for these fields as well? I tried to fetch some execution events for a rule using Postman and realized it's not that easy. I had to provide `start` and `end` dates in the correct ISO format (including milliseconds; without them, validation didn't pass), plus I had to adjust the dates to my local timezone. Having something like `now-1d` - `now` as the default date would make it a bit easier to request execution events.

Another good option here would be to return to date math expressions. They were more user-friendly, IMO 🙂

Reply:
Ahh, yeah, definitely agree there! For the alerting version of this API I believe they only made `startDate` required, and then would just calculate `endDate` as `n` number of executions based on the interval, and that's pretty nice. We could also do the reverse and default `endDate` to `now` and do the same for last `n` executions so it's always returning the most recent executions.

I would also like to bring back support for datemath strings as well, as they're just so easy to work/test with, so ++ there. Just need to be cautious of how it translates between UTC and the user's timezone. It should all be UTC and just formatted client side, so maybe a non-issue here....I just remember seeing some problem with this when testing awhile back but I could've been converting the datemath client-side which would've resulted in an issue for sure.

I'll capture this in an enhancement after I wrap up all the remaining items in the description 🙂

Reply:
Since this isn't immediately needed as part of https://github.com/elastic/kibana/pull/130072, I've created this enhancement https://github.com/elastic/kibana/issues/131095 for tracking.